README.md

```diff
@@ -56,6 +79,20 @@ There are a few things to note:
 Google's [protobuf](https://developers.google.com/protocol-buffers/docs/gotutorial) is a space efficient and fast format
 to serialize data in a portable way. It's easy to generate JSON out of
 
+## Tree Options
+### EnableHashSorting
+As described above, this is the flag to pass to implement a merkle tree with sorted hashes
+
+### SaltsLengthSuffix
```

Comment:
I am afraid that a user that has an already proto structure has a header level field that has the same name as that construction:
```example.proto:
string fieldA = 1;
repeated string fieldB = 2;
int fieldBLength = 3;
```

```example_salts.proto:
bytes fieldA = 1;
repeated bytes fieldB = 2;
bytes fieldBLength = 3;
```

That way there is one field that pick up the same salts as other one. If we can customize the suffix then that problem is gone. 
I agree that might be an edge case, but an ugly one if the user cannot rename their proto messages.

Reply:
Even in your example though it wouldn't be a problem because we use a `.` as a separator. Your example:

```
bytes fieldA = 1;
repeated bytes fieldB = 2;
bytes fieldBLength = 3;
```
would lead to the following leaves:

```
fieldA
fieldB[0]
fieldB[1]
fieldB.Length #NB: the .
fieldBLength
```

Reply:
There is no way you could have `fieldB.Length` as a valid property name.